File: scripts/download-product-images.js
#!/usr/bin/env node

const https = require('https');
const fs = require('fs');
const path = require('path');

// Read products data
const productsData = JSON.parse(fs.readFileSync(path.join(__dirname, '../best-system-v1/step2-top10list/products-data.json'), 'utf8'));

// Create products directory if it doesn't exist
const imagesDir = path.join(__dirname, '../src/assets/images/products');
if (!fs.existsSync(imagesDir)) {
  fs.mkdirSync(imagesDir, { recursive: true });
}

// Function to download image
function downloadImage(url, filename) {
  return new Promise((resolve, reject) => {
    const file = fs.createWriteStream(path.join(imagesDir, filename));
    
    https.get(url, (response) => {
      if (response.statusCode === 301 || response.statusCode === 302) {
        // Handle redirect
        https.get(response.headers.location, (redirectResponse) => {
          redirectResponse.pipe(file);
          file.on('finish', () => {
            file.close();
            resolve();
          });
        }).on('error', reject);
      } else {
        response.pipe(file);
        file.on('finish', () => {
          file.close();
          resolve();
        });
      }
    }).on('error', reject);
  });
}

// Function to get filename from URL
function getFilename(url) {
  const parts = url.split('/');
  return parts[parts.length - 1];
}

// Download all images
async function downloadAllImages() {
  console.log('🖼️  Downloading product images...\n');
  
  for (const product of productsData.products) {
    if (product.imageUrl && product.imageUrl.startsWith('http')) {
      const filename = getFilename(product.imageUrl);
      const localPath = path.join(imagesDir, filename);
      
      // Skip if already exists
      if (fs.existsSync(localPath)) {
        console.log(`✓ Already exists: ${filename}`);
        continue;
      }
      
      try {
        console.log(`⏬ Downloading: ${filename}...`);
        await downloadImage(product.imageUrl, filename);
        console.log(`✓ Downloaded: ${filename}`);
      } catch (error) {
        console.error(`❌ Failed to download ${filename}:`, error.message);
      }
    }
  }
  
  console.log('\n✅ Image download complete!');
  console.log('📁 Images saved to:', imagesDir);
}

// Run the download
downloadAllImages().catch(console.error);

File: scripts/download-product-images.py
#!/usr/bin/env python3
import json
import os
import requests
from urllib.parse import urlparse

# Read product data
with open('/home/titan/FactBench/src/data/pool-cleaners-complete.json', 'r') as f:
    data = json.load(f)

# Create images directory
images_dir = '/home/titan/FactBench/src/assets/images/products'
os.makedirs(images_dir, exist_ok=True)

# Download each product image
for product in data['products']:
    if 'imageUrl' in product:
        image_url = product['imageUrl']
        image_name = product['image']
        image_path = os.path.join(images_dir, image_name)
        
        print(f"Downloading {product['name']} image...")
        
        try:
            # Download image
            headers = {
                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
            }
            response = requests.get(image_url, headers=headers, stream=True)
            response.raise_for_status()
            
            # Save image
            with open(image_path, 'wb') as f:
                for chunk in response.iter_content(chunk_size=8192):
                    f.write(chunk)
            
            print(f"✅ Saved: {image_name}")
            
        except Exception as e:
            print(f"❌ Error downloading {image_name}: {e}")
            # Create placeholder note
            with open(image_path + '.txt', 'w') as f:
                f.write(f"Image URL: {image_url}\n")
                f.write(f"Error: {e}\n")

print("\n✅ Image download process completed!")
print(f"📁 Images saved to: {images_dir}")